pkg/simulator/core.go

```diff
@@ -15,26 +17,29 @@ import (
 // Simulation tracks all entities and activities related to device, host and link simulation
 type Simulation struct {
 	lock             sync.RWMutex
-	deviceSimulators map[simapi.DeviceID]*device.Simulator
-
-	// HostSimulators
-	// LinkSimulators
+	deviceSimulators map[simapi.DeviceID]*devsim.Simulator
+	linkSimulators   map[simapi.LinkID]*linksim.Simulator
+	hostSimulators   map[simapi.HostID]*hostsim.Simulator
 }
 
 // NewSimulation creates a new core simulation entity
 func NewSimulation() *Simulation {
 	return &Simulation{
-		deviceSimulators: make(map[simapi.DeviceID]*device.Simulator),
+		deviceSimulators: make(map[simapi.DeviceID]*devsim.Simulator),
```

Comment:
In ransim, we created simple data stores and an interface for each to make it easier to manage and interact with objects we want to create/get/update/delete. My suggestion is to separate stores from the actual logic but it is up to you.

Reply:
Possibly. This is equally to track the simulated entities and any simulation related processes/state. I'll leave it for now and may refactor later.